fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllTodos();
+  }
+
+  getAllTodos() {
+    /**
+     * Use fetch to send a GET request to `/api/todos`.
+     * Then 😉, once the response JSON is received and parsed,
+     * update state with the received todos.
+     */
+    fetch('/api/todos')
+      .then(res => res.json())
+      .then(data => this.setState({ todos: data }))
+  }
+
+  addTodo(newTodo) {
+    /**
+    * Use fetch to send a PO ST request to `/api/todos`.
+    * Then 😉, once the response JSON is received and parsed,
+    * add the created todo to the state array.
+    *
+    * TIP: Be sure to SERIALIZE the todo object in the body with JSON.stringify()
+    * and specify the "Content-Type" header as "application/json"
+    */
+   fetch('/api/todos', {
+     method: "POST",
+    body: JSON.stringify(newTodo),
+    headers: {"Content-type": "application/json; charset=UTF-8"}
+  })
+   .then(response => response.json())
+   .then(json => console.log(json))
```

Comment:
Here you need to update state with the new todo returned from the server.

```JSX
.then(todo => {
  // Make a copy of this.state.todos
  // Add todo to the copy of todos
  // Use the copy of todos to update the state
});
```

[How to copy an array in JS](https://www.samanthaming.com/tidbits/35-es6-way-to-clone-an-array/)